app/src/main/java/com/example/weather/base/BaseFragment.kt

```diff
@@ -25,7 +26,10 @@ abstract class BaseFragment<VB : ViewBinding>(private val inflate: FragmentInfla
 
     protected abstract fun bindData()
 
+    protected abstract fun checkNetwork(activity: Activity?)
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        checkNetwork(activity)
```

Comment:
dùng [Monitor connectivity status and connection metering ](https://developer.android.com/training/monitoring-device-state/connectivity-status-type) tạo 1 interface lắng nghe và implement nó như code sample dưới đây e nhé!

ref:
https://gist.github.com/Aeonitis/2e4df7a3b861edeb4b7d03e900bda633